pkg/logger/api.go

```diff
@@ -0,0 +1,56 @@
+package logger
+
+import (
+	"fmt"
+
+	"github.com/go-logr/logr"
+	"github.com/go-logr/zapr"
+	"go.uber.org/zap"
+)
+
+type ApiLogger struct {
+	logger logr.Logger
+}
+
+func NewApiLogger() Logger {
+	zap, err := zap.NewProduction()
+	if err != nil {
+		panic(err)
+	}
+	logger := zapr.NewLogger(zap)
+	return ApiLogger{
+		logger: logger,
+	}
+}
+
+func (l ApiLogger) Println(format string, a ...interface{}) {
+	l.logger.Info(fmt.Sprintf(format, a...))
+}
+
+func (l ApiLogger) Printf(format string, a ...interface{}) {
+	l.logger.Info(fmt.Sprintf(format, a...))
+}
+
+func (l ApiLogger) Actionf(format string, a ...interface{}) {
+	l.logger.Info(fmt.Sprintf(format, a...))
+}
+
+func (l ApiLogger) Generatef(format string, a ...interface{}) {
+	l.logger.Info(fmt.Sprintf(format, a...))
+}
+
+func (l ApiLogger) Waitingf(format string, a ...interface{}) {
+	l.logger.Info(fmt.Sprintf(format, a...))
+}
+
+func (l ApiLogger) Successf(format string, a ...interface{}) {
+	l.logger.Info(fmt.Sprintf(format, a...))
+}
+
+func (l ApiLogger) Warningf(format string, a ...interface{}) {
+	l.logger.Info(fmt.Sprintf(format, a...))
```

Comment:
Is it possible to suppress everything but errors and/or warnings with this logging package? I would expect to see `log.Warn` here.

I read through the README of this package and it seems they have some strong opinions on NOT doing `log.Warn`, `log.Debug`, etc.

Reply:
The approach is different as you've read, it uses numbers so there are no explicit log levels. And that's something I struggled with in this implementation, that's a bit hard to match. We can settle into something once we define the logging architecture.